oauth2_provider/models.py

```diff
@@ -148,7 +151,7 @@ def is_expired(self):
         return timezone.now() >= self.expires
 
     def redirect_uri_allowed(self, uri):
-        return uri == self.redirect_uri
+        return uri.split('?')[0] == self.redirect_uri.split('?')[0]
```

Comment:
You might want to use the `urlparse` (or Py3 `urllib.parse`) to do this kind of extraction.